Solutions/Menes.Specs/Steps/OpenApiHostingServiceCollectionExtensionsSteps.cs

```diff
@@ -0,0 +1,200 @@
+        }
+
+        [Then(@"the HalDocumentMapper for resource type has configured its links")]
+        public void ThenTheHalDocumentMapperForResourceTypeHasConfiguredItsLinks()
+        {
+            ServiceProvider provider = this.scenarioContext.Get<ServiceProvider>();
+            PetHalDocumentMapper mapper = provider.GetRequiredService<PetHalDocumentMapper>();
+            Assert.IsTrue(mapper.LinkMapConfigured);
+        }
+
+        [Then(@"the HalDocumentMapper for resource and context types has configured its links")]
+        public void ThenTheHalDocumentMapperForResourceAndContextTypesHasConfiguredItsLinks()
+        {
+            ServiceProvider provider = this.scenarioContext.Get<ServiceProvider>();
+            PetHalDocumentMapperWithContext mapper = provider.GetRequiredService<PetHalDocumentMapperWithContext>();
+            Assert.IsTrue(mapper.LinkMapConfigured);
+        }
+
+        [Then(@"the exception of type ""(.*)"" is mapped to response code ""(.*)""")]
+        public void ThenTheExceptionOfTypeIsMappedToResponseCode(string exceptionType, int statusCode)
+        {
+            IOpenApiExceptionMapper exceptionMapper = this.scenarioContext.Get<ServiceProvider>().GetRequiredService<IOpenApiExceptionMapper>();
+
+            // The only way to tell if it's been mapped without reflecting into the guts of the mapper is to try and register a new mapper for
+            // the given type/status code... even with this we need a little bit of reflection to invoke the method.
+            var type = Type.GetType(exceptionType);
+            MethodInfo mapMethod = exceptionMapper.GetType().GetMethods().First(x => x.Name == "Map" && x.GetGenericArguments().Length == 1);
+            MethodInfo createdMapMethod = mapMethod.MakeGenericMethod(type);
+
+            try
+            {
+                createdMapMethod.Invoke(exceptionMapper, new object[] { statusCode, null });
+
+                Assert.Fail($"Exception of type '{exceptionType}' was not registered");
+            }
+            catch (TargetInvocationException ex)
+            {
+                // The exception will have been wrapped in a TargetInvocationException because we invoked via reflection, so we need to
+                // see if the inner exception was what we expected.
+                if (!(ex.InnerException is ArgumentException))
+                {
+                    throw;
+                }
+            }
+        }
+
+        [Then(@"an audit log builder service is added for auditing operations which return OpenApiResults")]
+        public void ThenAnAuditLogBuilderServiceIsAddedForAuditingOperationsWhichReturnOpenApiResults()
+        {
+            this.AssertServiceIsRegistered<IAuditLogBuilder, OpenApiResultAuditLogBuilder>();
+        }
+
+        [Then(@"an audit log builder service is added for auditing operations which return a POCO")]
+        public void ThenAnAuditLogBuilderServiceIsAddedForAuditingOperationsWhichReturnAPOCO()
```

Comment:
Not a major one this...but conventionally in .NET, acronyms get CamelCased, so POCO should become `Poco` in a method name.